### PR TITLE
[wasm][debugger] Fixing error after receiving loaded_files = null

### LINKED ISSUE #39301: Blazor Web Assembly - Debugger crashes on VS 2022 after OIDC redirect
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When pressing F5 on a Blazor WASM application, after the OIDC redirection to the login page of the IdP the application crashes throwing the following exception,

### Expected Behavior

Debugger working as normally.
Application not crashing.

### Steps To Reproduce

_No response_

### Exceptions (if any)

```
fail: Microsoft.WebAssembly.Diagnostics.DevToolsProxy[0]::Run:
 Exception System.AggregateException: One or more errors occurred.
```

### .NET Version

.NET 6

### Anything else?

Visual Studio 2022
Version 17.0.4

### PR DESCRIPTION
**Pull request by thaystg**
Added a test case that cannot reproduce the same as the user is getting but at least we can see the error being logged correctly.

Probably fixes https://github.com/dotnet/aspnetcore/issues/39301

### CODE REVIEW

File: src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs [hunk 1]
```diff
@@ -707,5 +707,76 @@ public async Task DebuggerAttributeStopOnDebuggerHiddenCallWithDebuggerBreakCall
...
+                expression = "window.setTimeout(function() { reload_wasm_page(); }, 1);"
+            });
+            await cli.SendCommand("Runtime.evaluate", run_method, token);
+            await Task.Delay(1000, token);
+            await insp.WaitFor(Inspector.READY);
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 10, 8,
+                "IntAdd",
+                wait_for_event_fn: (pause_location) =>
```
Comment (radical) [inline src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs:756]:
```suggestion
                wait_for_event_fn: async (pause_location) =>
```

File: src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	<head>
+	</head>
+	<body>
+	<script type='text/javascript'>
+		var App = {
+			init: function () {
+				MONO.loaded_files = null;
+			},
+		};
+        function reload_wasm_page () {
```
Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html:12]: space->tabs

File: src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			window.location.replace("http://localhost:9400/debugger-driver.html");
+		}
```
Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html:14]:
```suggestion
		}
```

File: src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html (hunk 2 above)
Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html]: done

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs [hunk 4]
```diff
@@ -1263,13 +1263,19 @@ internal async Task<DebugStore> LoadStore(SessionId sessionId, CancellationToken
                 if (loaded_files == null)
                 {
                     Result loaded = await SendMonoCommand(sessionId, MonoCommands.GetLoadedFiles(RuntimeId), token);
-                    loaded_files = loaded.Value?["result"]?["value"]?.ToObject<string[]>();
+                    if (loaded.IsOk)
+                        loaded_files = loaded.Value?["result"]?["value"]?.ToObject<string[]>();
+                    else
+                        Log("warning", $"Error on mono_wasm_get_loaded_files {loaded.Value}");
```
[the remark this answers is not in the document]
Comment (thaystg) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs]: don

**Review by radical — APPROVED** [review]

**Comment by thaystg** [comment]
@lewing can I backport to 6.0?

**Pull request merged**